src/lock.rs

```diff
@@ -0,0 +1,328 @@
+
+const SHA256_TAG: &str = "sha256";
+
+#[derive(Clone, PartialEq, Eq)]
+pub enum DigestAlgorithm {
+    SHA256,
+}
+
+impl std::fmt::Display for DigestAlgorithm {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let s = match self {
+            Self::SHA256 => SHA256_TAG,
+        };
+        f.write_str(s)
+    }
+}
+
+impl FromStr for DigestAlgorithm {
+    type Err = eyre::Report;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            SHA256_TAG => Ok(Self::SHA256),
+            other => eyre::bail!("invalid digest algorithm: {other}"),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct Digest {
+    bytes: Vec<u8>,
+    algorithm: DigestAlgorithm,
+}
+
+impl TryFrom<digest::Digest> for Digest {
+    type Error = eyre::Report;
+
+    fn try_from(value: digest::Digest) -> Result<Self, Self::Error> {
+        let algorithm = if value.algorithm() == &digest::SHA256 {
+            DigestAlgorithm::SHA256
+        } else {
+            eyre::bail!("Unsupported digest algorithm: {:?}", value.algorithm())
+        };
+
+        Ok(Self {
+            bytes: value.as_ref().to_vec(),
+            algorithm,
+        })
+    }
+}
+
+impl std::fmt::Display for Digest {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!(
+            "{}:{}",
+            self.algorithm,
+            hex::encode(&self.bytes)
+        ))
+    }
+}
+
+impl Serialize for Digest {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        self.to_string().serialize(serializer)
+    }
+}
+
+struct DigestVisitor;
+
+impl<'de> Visitor<'de> for DigestVisitor {
+    type Value = Digest;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        formatter.write_str("a hexadecimal encoded cryptographic digest")
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        let mut parts = value.split(':');
+        let algorithm_tag = parts.next().ok_or(E::missing_field("algorithm"))?;
+        let algorithm = algorithm_tag
+            .parse::<DigestAlgorithm>()
+            .map_err(|_| E::custom("invalid digest algorithm"))?;
+        let bytes = parts
+            .next()
+            .ok_or(E::missing_field("bytes"))
+            .and_then(|h| hex::decode(h).map_err(|_| E::custom("invalid encoding")))?;
+        Ok(Self::Value { algorithm, bytes })
+    }
+}
+
+impl<'de> Deserialize<'de> for Digest {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        deserializer.deserialize_str(DigestVisitor)
+    }
+}
+
+impl PartialEq for Digest {
+    fn eq(&self, other: &Self) -> bool {
+        self.algorithm == other.algorithm && self.bytes == other.bytes
+    }
+}
+
+impl Eq for Digest {}
```

Comment:
I think these can be derived.

Reply:
Oh good catch, this is leftover from a previous version of the Digest representation.